carbon-p2-plugin/src/main/java/org/wso2/maven/p2/feature/generate/utils/FeatureFileGeneratorUtils.java

```diff
@@ -0,0 +1,452 @@
+                } else {
+                    plugin.setAttribute("match", P2Utils.getMatchRule(feature.getCompatibility()));
+                }
+                require.appendChild(plugin);
+            }
+        }
+
+        if (includedFeatures != null) {
+            for (Feature includedFeature : includedFeatures) {
+                Element includeElement = document.createElement("includes");
+                includeElement.setAttribute("id", includedFeature.getId());
+                includeElement.setAttribute("version", includedFeature.getVersion());
+                includeElement.setAttribute("optional", Boolean.toString(includedFeature.isOptional()));
+                rootElement.appendChild(includeElement);
+            }
+        }
+
+        for (Feature feature : missingImportFeatures) {
+            if (feature.isOptional()) {
+                Element includeElement = document.createElement("includes");
+                includeElement.setAttribute("id", feature.getId());
+                includeElement.setAttribute("version", feature.getVersion());
+                includeElement.setAttribute("optional", Boolean.toString(feature.isOptional()));
+                rootElement.appendChild(includeElement);
+            }
+        }
+        //endregion
+
+        try {
+            TransformerFactory transformerFactory = TransformerFactory.newInstance();
+            Transformer transformer;
+            transformer = transformerFactory.newTransformer();
+            DOMSource source = new DOMSource(document);
+            StreamResult result = new StreamResult(featureXmlFile);
+            transformer.setOutputProperty(OutputKeys.INDENT, "yes");
+            transformer.setOutputProperty("{http://xml.apache.org/xslt}indent-amount", "2");
+            transformer.transform(source, result);
+        } catch (TransformerConfigurationException e) {
+            throw new TransformerConfigurationException("Unable to create feature manifest", e);
+        }
+    }
+
+    /**
+     * If a manifest file is given, parse the manifest file and return the Document object representing the file.
+     * Generates a new Document otherwise.
+     *
+     * @param manifest java.io.File pointing an existing manifest file.
+     * @return Document object representing a given manifest file or a newly generated manifest file
+     * @throws ParserConfigurationException
+     * @throws SAXException
+     * @throws IOException
+     */
+    private static Document getManifestDocument(File manifest) throws ParserConfigurationException,
+            SAXException, IOException {
+        DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder documentBuilder;
+        try {
+            documentBuilder = documentBuilderFactory.newDocumentBuilder();
+        } catch (ParserConfigurationException e) {
+            throw new ParserConfigurationException("Unable to load feature manifest");
```

Comment:
ParserConfigurationException class has only 2 types of constructors. A default constructor and a constructor which take one string parameter. Thus it is not possible to pass the stack trace. Rather than throwing the caught ParserConfigurationException, a new exception with a more meaningful message is thrown from the method.